### PR TITLE
splunk improve

### PR DESCRIPTION
**Pull request by esharf**
## Status
- [x] Ready

## Related Issues
related: https://github.com/demisto/etc/issues/34906
fixes: https://github.com/demisto/etc/issues/35550

## Minimum version of Cortex XSOAR
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests

### CODE REVIEW

File: Packs/SplunkPy/ReleaseNotes/2_1_12.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### SplunkPy
+- Updated the Docker image to: *demisto/splunksdk:1.0.0.24033*.
+- Maintenance and stability enhancements.
```
Comment (yaakovi) [inline Packs/SplunkPy/ReleaseNotes/2_1_12.md:5]: specify the changes made to the integration

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 2]
```diff
@@ -2162,6 +2168,25 @@ def get_store_data(service):
         yield store.data.query(**query)
 
 
+class MessagesHandler:
+
+    service = None
+
+    @staticmethod
+    def get_messages_data(notable_id):
+        if MessagesHandler.service is not None:
```
Comment (yaakovi) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2177]: this way you can decrease the area under the indent (for the rest of the function)
```suggestion
        if MessagesHandler.service is None:
            return []
```

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 3]
```diff
@@ -2162,6 +2168,25 @@ def get_store_data(service):
         yield store.data.query(**query)
 
 
+class MessagesHandler:
```
Comment (yaakovi) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2171]: why do you need a class for it? why can't it just be a regular function that gets the service as an argument?

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 4]
```diff
@@ -2162,6 +2168,25 @@ def get_store_data(service):
...
+    @staticmethod
+    def get_messages_data(notable_id):
+        if MessagesHandler.service is not None:
+            comments = []
+            query = '|`incident_review` | where rule_id="{}" | search comment=* reviewer=*'.format(notable_id)
+            for comment in json.load(MessagesHandler.service.jobs.oneshot(query, output_mode='json'))['results']:
+                comments.append({
+                    'comment': comment['comment'],
+                    'reviewer': comment['reviewer'],
+                    'time': comment['_time']
```
Comment (yaakovi) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2184]:
```suggestion
                    'time': comment['_time'],
```

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 5]
```diff
@@ -292,6 +292,9 @@ def create_incident(notable_data, occurred):
 
         incident["occurred"] = occurred
         notable_data = parse_notable(notable_data)
+        notable_id = notable_data.get('rule_id')
```
Comment (guykeller) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:295]: Why is this called a notable ID? can you explain?

**Review by guykeller — COMMENTED** [review]
Let's make sure we add necessary changes to the Splunk prerelease version as well

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py (hunk 4 above)
Comment (esharf) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py]: Fixed

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py (hunk 2 above)
Comment (esharf) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py]: Fixed

File: Packs/SplunkPy/ReleaseNotes/2_1_12.md (hunk 1 above)
Comment (esharf) [reply Packs/SplunkPy/ReleaseNotes/2_1_12.md]: Fixed

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py (hunk 3 above)
Comment (esharf) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2171]: we discussed it and decided that it’s OK in this situation.

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py (hunk 5 above)
Comment (esharf) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:295]: It’s the notable id on the Splunk side.

**Comment by guykeller** [comment]
@esharf 
Did you see [this comment](https://github.com/demisto/content/pull/14713#pullrequestreview-755696108)?

**Comment by esharf** [comment]
> @esharf Did you see [this comment](https://github.com/demisto/content/pull/14713#pullrequestreview-755696108)?

Sorry missed it.
fixed now.

**Comment by okaufman34** [comment]
@yaakovi  @guykeller  Doc review completed

File: Packs/SplunkPyPreRelease/Integrations/SplunkPyPreRelease/SplunkPyPreRelease.py [hunk 6]
```diff
@@ -2338,6 +2345,26 @@ def get_store_data(service):
...
 
+class MessagesHandler:
+
+    service = None
+
+    @staticmethod
+    def get_messages_data(notable_id):
+        if MessagesHandler.service is None:
+            return []
+        comments = []
```
Comment (yaakovi) [inline Packs/SplunkPyPreRelease/Integrations/SplunkPyPreRelease/SplunkPyPreRelease.py:2356]:
```suggestion

        comments = []
```

File: Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        **get_command_args(args)
+    }
+
+    if 'incident_id' in args:
+        demisto.executeCommand("executeCommandAt", {"command": "splunk-notable-event-edit",
+                               "incidents": args['incident_id'], "arguments": command_args})
+    else:
+        demisto.executeCommand("splunk-notable-event-edit", command_args)
+
+    demisto.results('ok')
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py:51]: improve the returned message
```suggestion
    return_results('ok')
```

File: Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    command_args = {
+        'comment': args['comment'],
+        **get_command_args(args)
+    }
+
+    if 'incident_id' in args:
+        demisto.executeCommand("executeCommandAt", {"command": "splunk-notable-event-edit",
+                               "incidents": args['incident_id'], "arguments": command_args})
+    else:
+        demisto.executeCommand("splunk-notable-event-edit", command_args)
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py:49]: use `execute_command` instead (it contains error handling)

File: Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+def get_notable_data_from_incident(incident_id: str) -> Dict[str, str]:
+    """
+    Args:
+        incident_id(str): The identifier of the incident in XSOAR system.
+
+    Returns:
+        a dict of the notable data required to run the edit event comand.
+    """
+    try:
+        incidents_data = demisto.executeCommand("getIncidents", {'id': incident_id})[0]['Contents']['data'][0]
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py:16]: use `execute_command` instead

File: Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        arguments = {"eventIDs": incidents_data['dbotMirrorId']}
+        instance = incidents_data.get('sourceInstance')
+        if instance:
+            arguments["using"] = instance
+
+        return arguments
+    except (KeyError, IndexError):
+        raise DemistoException('cuold not determine the notable id and mirror instance.') from None
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.py:24]: log the original error and pass it to the DemistoException as `exception`

File: Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.yml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+enabled: false
+name: SplunkCommentNotableEvent
+script: '-'
+system: false
+tags: []
+timeout: '0'
+type: python
+subtype: python3
+dockerimage: demisto/python3:3.9.7.24076
+fromversion: 5.0.0
```
Comment (yaakovi) [inline Packs/SplunkPy/Scripts/SplunkCommentNotableEvent/SplunkCommentNotableEvent.yml:28]:
```suggestion
fromversion: 6.0.0
```

File: Packs/SplunkPy/ReleaseNotes/2_2_1.md [hunk 12]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+##### SplunkPy
+- Added detailed comments section in the raw response.
+
+#### Scripts
+##### New: SplunkCommentNotableEvent
+- Add a comment to a Splunk notable event. (Available from Cortex XSOAR 5.0.0).
```
Comment (yaakovi) [inline Packs/SplunkPy/ReleaseNotes/2_2_1.md:8]:
```suggestion
- Add a comment to a Splunk notable event. (Available from Cortex XSOAR 6.0.0).
```

**Pull request closed without merging**